contrib_bots/lib/github_issues.py

```diff
@@ -0,0 +1,123 @@
+    an item is prefixed with '@issue' or '@bug'
+
+    It will also write items to the issues stream, as well
+    as reporting it to github
+    '''
+
+    def usage(self):
+        return '''
+            This plugin will allow users to flag messages
+            as being issues with Zulip by using te prefix '@issue'
+
+            Before running this, make sure to create a stream
+            called "issues" that your API user can send to.
+
+            Also, make sure that the credentials of the github bot have
+            been typed in correctly, that there is a personal access token
+            with access to public repositories ONLY,
+            and that the repository name is entered correctly.
+
+            Check ~.zuliprc-local, and make sure there are
+            github_repo
+            github_repo_owner
+            github_username
+            '''
+
+    def triage_message(self, message):
+        # return True if we want to (possibly) respond to this message
+
+        original_content = message['content']
+
+        # This next line of code is defensive, as we
+        # never want to get into an infinite loop of posting follow
+        # ups for own follow ups!
+        if message['display_recipient'] == 'issue':
+            return False
+        is_issue = (original_content.startswith('@issue'))
+
+        return is_issue
+
+    def handle_message(self, message, client, state_handler):
+        original_content = message['content']
+        original_sender = message['sender_email']
+
+        # gets token from dev-secrets
+        # CONFIGURE TOKEN ON LOCAL MACHINE FOR NOW AT .zuliprc-local
+        config = ConfigParser.ConfigParser()
+        config.read([os.path.expanduser('~/zulip/zproject/dev-secrets.conf')])
+        oauth_token = config.get('secrets', 'github_issue_bot_token')
+
+        # Authentication for user filing issue DOES NOT WORK IF
+        # THE USER HAS READ/WRITE ACCESS TO THE REPOSITORY
+        config.read([os.path.expanduser('~/.zuliprc-local')])
+        USERNAME = config.get('api', 'github_username')
+
+        # The repo name to post to
+        REPO_NAME = config.get('api', 'github_repo')
+
+        # The owner of the repo to post to
+        REPO_OWNER = config.get('api', 'github_repo_owner')
+
+        new_content = original_content.replace('@issue',
+                                          'by %s:' % (original_sender,))
+        # gets the repo url
+        url = 'https://api.github.com/repos/%s/%s/issues' % (REPO_OWNER, REPO_NAME)
+
+        # signs into github using the provided username and password
+        session = requests.session()
+        session.auth = (USERNAME, oauth_token)
+
+        # Gets rid of the @issue in the issue title
+        issue_title = message['content'].replace('@issue', '').strip()
+        issue_content = ''
+        cut_string = False
+        cut_len = 0
+        for i in range(0, len(issue_title)):
```

Comment:
I think you could change it to:
`for index, character in enumerate(issue_title)):`
then instead of `issue_title[i]` you could just write `character`

Sounds better?